packages/validator/src/validator.ts

```diff
@@ -265,6 +294,40 @@ export class Validator {
       return BeaconHealth.ERROR;
     }
   }
+
+  private async fetchValidatorStatuses(pubkeysHex: string[], metrics: Metrics | null): Promise<void> {
+    const res = await this.api.beacon.getStateValidators("head", {
+      id: pubkeysHex,
+    });
+    ApiError.assert(res, "Can not fetch state validators from beacon node");
+
+    const allValidatorStatuses = new MapDef<SimpleValidatorStatus, number>(() => 0);
+
+    for (const validatorState of res.response.data) {
+      // Group all validators by status
```

Comment:
does this comment make sense inside the for-loop?

Reply:
it could be outside the loop as well, indifferent to whether it improves readability.

Reply:
doesn't matter that much but I would probably put it outside as the combined iterations "group all validators by status" and not a single iteration